Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/DPEComponentAdapter.cpp

```diff
@@ -141,19 +142,19 @@ namespace AZ::DocumentPropertyEditor
         return returnValue;
     }
 
-    void ComponentAdapter::OnBeginRow(AdapterBuilder* adapterBuilder, AZStd::string_view serializedPath)
+    void ComponentAdapter::CreateLabel(AdapterBuilder* adapterBuilder, AZStd::string_view labelText, AZStd::string_view serializedPath)
     {
-        if (!serializedPath.empty())
+        auto* prefabAdapterInterface = AZ::Interface<AzToolsFramework::Prefab::PrefabAdapterInterface>::Get();
+        if (prefabAdapterInterface)
```

Comment:
Discussed offline that binding prefab with DPEComponentAdapter is fine. But we should avoid depending on prefab system in classes like EntityPropertyEditor, ComponentEditor, ReflectionAdapter, DocumentPropertyEditor. For integrating UI editor with DPE, we would just use a different class or derive DPEComponentAdapter to meet our needs.

After this PR, we discussed that rather than having a PrefabAdapter singleton we can try to establish parent-child relationship between PrefabAdapter and DPEComponentAdapter.

Reply:
To me, it seems like an unnecessary dependency to put prefab stuff in AzToolsFramework::DocumentPropertyEditor files. I would argue that we should be deriving from DPEComponentAdapter for the Entity Property Editor, and leaving the default without prefab references.

Reply:
Discussed offline again. Will address in a separate PR so that a custom adapter can be assigned to ComponentEditor.